### PR TITLE
Use bleach to strip out tags that dont exist in our whitelist

### PR DESCRIPTION
**Pull request by anurag90x**
Structural validation is more complex and harder to extend and maintain vs a whitelist based approach. Hence, the switch to bleach.

### CODE REVIEW

File: markup_language/schema_validator.py [hunk 1]
```diff
@@ -1,19 +1,34 @@
...
+
+def get_sanitized_document(xml_string):
+    unescaped_string = unescape(xml_string.encode('utf-8'))
+    clean_content = bleach.clean(
+        unescaped_string, tags=tags,
+        attributes=attributes, styles=styles,
+        protocols=allowed_protocols
+    )
+    clean_content = bleach.linkify(clean_content,
+                                   callbacks=[add_rel_attributes])
```
Comment (markstory) [inline markup_language/schema_validator.py]: Won't this add links to any URLs in the source text? Is that what customer's using a wysiwyg will expect?

Comment (anurag90x) [inline markup_language/schema_validator.py]: I can make it add it to only the explicit links

File: markup_language/schema_validator.py [hunk 2]
```diff
@@ -1,19 +1,40 @@
...
+    unescaped_string = unescape(xml_string.encode('utf-8'))
+    url_regex = URL_RE if linkify else re.compile("$^")
+    linkifier = partial(LinkifyFilter,
+                        callbacks=[add_rel_attributes], url_re=url_regex)
+    clean_content = Cleaner(
+        tags=tags,
+        attributes=attributes, styles=styles,
+        protocols=allowed_protocols,
+        filters=[linkifier]
+    ).clean(unescaped_string)
```
Comment (markstory) [inline markup_language/schema_validator.py]: Are we still constraining the source of images?

Comment (anurag90x) [inline markup_language/schema_validator.py]: We are now!

File: markup_language/schema_validator.py [hunk 3]
```diff
@@ -1,19 +1,65 @@
...
-    dir = os.path.dirname(__file__)
-    file_name = os.path.join(dir, 'schema.xsd')
-    with open(file_name, 'r') as f:
-        schema_root = etree.XML(f.read())
-        return etree.XMLSchema(schema_root)
+def allowed_image_attrs(tag, name, value):
+    if name in ['alt', 'class', 'height', 'width', 'style']:
+        return True
+    if name == 'src':
+        path = urlparse(value).path
```
Comment (markstory) [inline markup_language/schema_validator.py]: Can `urlparse` raise an error?

Comment (anurag90x) [inline markup_language/schema_validator.py]: Given a string it always returns an object

**Pull request merged**